tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs

```diff
@@ -2,84 +2,157 @@
 
     [ObservableProperty]
-    private bool isPackaged = false;
+    private bool _isRunningAsSystem = false;
 
     [ObservableProperty]
-    private bool isStoreApp = false;
+    private Visibility _visibility = Visibility.Visible;
 
     [ObservableProperty]
-    private bool isRunningAsAdmin = false;
+    private string _activationArgs = string.Empty;
 
     [ObservableProperty]
-    private bool isRunningAsSystem = false;
+    private PackageInfo _packageInfo = new();
 
     [ObservableProperty]
-    private Visibility visibility = Visibility.Visible;
+    private string _identifiedFrameWorkTypes = string.Empty;
 
-    public ObservableCollection<FrameworkType> FrameworkTypes { get; } = [];
+    private List<FrameworkType> FrameworkTypes { get; } = [];
 
     public AppRuntimeInfo()
     {
-        // Note these are in alphabetical order of Name.
         FrameworkTypes.Add(new FrameworkType("Avalonia.Base.dll", "Avalonia"));
         FrameworkTypes.Add(new FrameworkType("DXCore.dll", "DirectX"));
         FrameworkTypes.Add(new FrameworkType("Microsoft.Maui.dll", "Maui"));
         FrameworkTypes.Add(new FrameworkType("MFC", "MFC", false));
         FrameworkTypes.Add(new FrameworkType("Python.exe", "Python"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.Web.WebView2.Core.dll", "WebView2"));
         FrameworkTypes.Add(new FrameworkType("Microsoft.Windows.SDK.NET.dll", "Windows App SDK"));
         FrameworkTypes.Add(new FrameworkType("System.Windows.Forms.dll", "Windows Forms"));
-        FrameworkTypes.Add(new FrameworkType("Microsoft.WinUI.dll", "WinUI"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.UI.Xaml.dll", "WinUI 2"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.UI.Xaml.Controls.dll", "WinUI 3"));
         FrameworkTypes.Add(new FrameworkType("PresentationFramework.dll", "WPF"));
     }
 
-    public void CheckFrameworkTypes(string moduleName)
+    public void CheckFrameworksAndCommandLine(Process process)
     {
-        foreach (var item in FrameworkTypes)
+        var identifiedFrameworks = new List<string>();
+        var modules = process.Modules;
+        foreach (ProcessModule module in modules)
         {
-            // Skip if already matched.
-            if (item.IsTypeSupported == true)
+            for (var i = 0; i < FrameworkTypes.Count; i++)
             {
-                continue;
-            }
+                var item = FrameworkTypes[i];
 
-            if (item.IsExactMatch)
-            {
-                if (moduleName.Equals(item.Determinator, StringComparison.OrdinalIgnoreCase))
+                if (identifiedFrameworks.Contains(item.Name))
                 {
-                    item.IsTypeSupported = true;
+                    continue;
                 }
-            }
-            else
-            {
-                if (moduleName.Contains(item.Determinator, StringComparison.OrdinalIgnoreCase))
+
+                if (item.IsExactMatch)
```

Comment:
If all you're doing when item.IsExactMatch is true is using string.equals, can you get rid of the first if?

    if(name.Equals(...))
    {
    }
    else if(name.Contains(...))
    {
    }

Reply:
The IsExactMatch property indicates that an exact match is required, hence the use of Equals; whereas if not required then I must use Contains instead. I can't use Contains if IsExactMatch==true.